audiofilelibrary.go

```diff
@@ -19,74 +19,83 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 package main
 
 import (
-	"bufio"
+	"errors"
+	"io/ioutil"
 	"log"
-	"os"
+	"path/filepath"
+	"strconv"
 	"strings"
 )
 
+// AudioFileLibrary is a structure to hold info on the files in the audio directory.
 type AudioFileLibrary struct {
-	filePath  string
-	FilesById map[string]string
+	soundsDirectory string
+	FilesByID       map[int]string
 }
 
-func OpenLibrary(filePath string) *AudioFileLibrary {
-	return (&AudioFileLibrary{}).openLibrary(filePath)
-}
-
-func (library *AudioFileLibrary) openLibrary(filePath string) *AudioFileLibrary {
-	library.filePath = filePath
-	library.FilesById = make(map[string]string)
+// Take a file name and extract an ID from it.
+// The file names are similar to "bellbird-6.mp3".  But they could be like "SI Kaka-SI Kaka-17.mp3"
+// Need to handle cases like "schedule.json" i.e. the file is not an audio file.
+func extractIDFromFileName(fileName string) (int, error) {
 
-	// Open the file and scan it.
-	f, err := os.Open(filePath)
-	if err != nil {
-		log.Printf("Error loading audio library %s", err)
-		return library
+	lastIndex := strings.LastIndex(fileName, "-")
+	if lastIndex < 0 {
+		return -1, errors.New("Skipping file with name " + fileName)
 	}
 
-	scanner := bufio.NewScanner(f)
-	for scanner.Scan() {
-		line := scanner.Text()
-
-		if len(line) > 0 && line[0] != '%' {
-			parts := strings.SplitN(line, ":", 2)
-
-			if len(parts) > 1 {
-				library.FilesById[parts[0]] = strings.Trim(parts[1], " ")
-			}
-		}
-	}
-	if scanner.Err() != nil {
-		log.Printf("Error loading audio library %s", scanner.Err())
+	fileIDWithExtension := fileName[lastIndex+1:]
+	fileIDStr := strings.TrimSuffix(fileIDWithExtension, filepath.Ext(fileIDWithExtension))
+	fileID, err := strconv.Atoi(fileIDStr)
+	if err != nil {
+		return -1, err
 	}
 
-	return library
+	return fileID, nil
 }
 
-func (library *AudioFileLibrary) AddFile(fileId, filename string) error {
-	firstItem := len(library.FilesById) == 0
+// OpenLibrary reads the audio directory.  And constructs a map of file IDs to file names.
+func OpenLibrary(soundsDirectory string) *AudioFileLibrary {
 
-	library.FilesById[fileId] = filename
+	log.Println("Reading audio directory")
+	library := &AudioFileLibrary{
+		soundsDirectory : soundsDirectory,
+		FilesByID : make(map[int]string),
+	}
 
-	f, err := os.OpenFile(library.filePath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	files, err := ioutil.ReadDir(soundsDirectory)
 	if err != nil {
-		return err
+		log.Println("Error reading audio directory", err)
+		return library
```

Comment:
Instead of logging the error and returning an empty library, this should return `nil, err` and let the caller deal with the error.